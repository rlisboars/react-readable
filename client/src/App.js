import React, { Component } from 'react'
import PostsList from './components/PostsList'
import CommentsList from './components/CommentsList'
import { Route, Switch } from 'react-router-dom'
 
export class App extends Component {

	render() {
		return (
			<div className="App">
				<Switch>
				<Route path='/' exact component={PostsList} />
				<Route path='/:category/:id' component={CommentsList} />
				<Route path='/:category' component={PostsList} />
				</Switch>
				<footer>
					<div className='spacer' />
					<div className='footer'>
						by <a href="https://github.com/rlisboars/" target="_blank" rel="noopener noreferrer" style={{'marginBottom': '0px'}}>rlisboars</a>
					</div>
				</footer>
			</div>
		);
	}
}

export default App;
