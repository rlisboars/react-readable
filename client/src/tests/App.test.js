import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-15'

describe('App Component testing', () => {
    it('##### renders without crashing', () => {
        expect(shallow(<App/>).length).toEqual(1)
    });
    it('#### capturing snapshot of App', () => {
        const wrapper = shallow(<App/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

