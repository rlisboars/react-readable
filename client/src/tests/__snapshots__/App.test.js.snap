// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component testing #### capturing snapshot of App 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/:category/:id"
    />
    <Route
      component={[Function]}
      path="/:category"
    />
  </Switch>
  <footer>
    <div
      className="spacer"
    />
    <div
      className="footer"
    >
      by 
      <a
        href="https://github.com/rlisboars/"
        rel="noopener noreferrer"
        style={
          Object {
            "marginBottom": "0px",
          }
        }
        target="_blank"
      >
        rlisboars
      </a>
    </div>
  </footer>
</div>
`;
